account_sale_stock_report_non_billed/models/stock_move.py

```diff
@@ -127,6 +127,8 @@ def _compute_not_invoiced_values(self):
             moves_in_date = invoices_not_cancel.mapped("move_line_ids").filtered(
                 lambda m: m.date_done >= date_start and m.date_done <= date_end
             )
+            if not self.env.context.get("date_check_iterval_restrict", False):
+                date_start = False
```

Comment:
I don't get why do you need this. Pass the minimum variables in context, and the absence of one of them means that no check should be done.

Reply:
Because we need the date_start and the date_end to filter the move_line_ids in any situation, but when the user doesn't want to use this dates as a range for invoices too the date_start has to be reset to False.

Reply:
Yes, and that's why I say you only need to pass one variable: the start date. If not passed, then you use `.get(..., False)`, and being `False` means that no check has to be done.

Reply:
But I need the date_start to take the `moves_in_date`

Reply:
OK, I'm getting it. They should be independent variables for avoiding that clunky logic. Pass these 3 possible variables to context:

- `non_billed_date`: Old `date_check_invoiced_moves`
- `non_billed_date_start`: Old `date_check_invoiced_moves_start`
- `non_billed_invoice_date_start`: This one is set to `self.stock_move_non_billed_threshold` if `self.interval_restrict_invoices`.

and you get each of them on the computing code, with no inter-dependencies between them.